colossus/src/main/scala/colossus/controller/Pipe.scala

```diff
@@ -99,60 +189,132 @@ trait Source[+T] extends Transport {
 
   def collected: Callback[Iterator[T]] = fold(new collection.mutable.ArrayBuffer[T]){ case (next, buf) => buf append next ; buf } map {_.toIterator}
 
+  /**
+   * Link this source to a sink.  Items will be pulled from the source and
+   * pushed to the sink, respecting backpressure, until either the source is
+   * closed or an error occurs.  The sink will be closed when this source is
+   * closed.  If the sink is closed before this source, this source will be
+   * terminated.  Other terminations are propagated in both directions.
+   */
+  def into[U >: T] (sink: Sink[U]) {
+    def tryPush(item: T): Boolean = sink.push(item) match {
+      case PushResult.Filled(sig) => {
```

Comment:
Nitpick: I'm guessing `sig` means `signal`? Being more explicit with variable names makes it easier to read.